src/nunavut/lang/cpp/support/variable_length_array.hpp

```diff
@@ -1413,6 +1413,7 @@ class VariableLengthArray<bool, MaxSize, Allocator>
 #endif
         if (new_data != nullptr)
         {
+            std::fill_n(new_data, bits2bytes(no_shrink_capacity), 0);  // Avoid uninitialized reads during modification
```

Comment:
I'm pretty sure this is wrong (I'm also surprised it doesn't break the tests). If you fill this pointer but the allocator had extended the existing memory then you just blew away the contents of the variant.

Reply:
You are right. Fixed in the follow-up commit.